docs/source/model/signatures.rst

```diff
@@ -0,0 +1,852 @@
+
+
+Signature Playground
+^^^^^^^^^^^^^^^^^^^^
+
+.. toctree::
+    :maxdepth: 1
+    :hidden:
+
+    notebooks/signature_examples
+
+In order to aid in understanding how data structures will be automatically inferred as a valid signature, as well as to provide extensive examples of 
+valid signatures, we've created a notebook that you can view, showing different examples and their resulting inferred signatures. 
+You can `view the notebook here <notebooks/signature_examples.html>`_. 
+
+Alternatively, if you'd like to download the notebook locally and test it out with your own data structures, you can download it below.
+
+.. raw:: html
+
+    <a href="https://raw.githubusercontent.com/mlflow/mlflow/master/docs/source/model/notebooks/signature_examples.ipynb" class="notebook-download-btn">
+        <i class="fas fa-download"></i>Download the Signature Playground Notebook</a><br/>
+
+Required vs. Optional Input fields
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There are certain conditions that determine enforcement of signatures that are important to consider when defining your model signatures. One of the 
+most notable is the concept of required vs. optional with regards to input data. 
+
+Required fields are those that must be present in the input data in order for the model to be able to make a prediction. If a required field is missing, 
+the signature enforcement validation will raise an exception stating that the required input field is missing. 
+
+In order to configure a field as optional, you must pass in a value of `None` or `np.nan` for that field when using the :py:func:`mlflow.models.infer_signature` 
+function. Alternatively, you can manually define the signature and set the `required` field to `false` for that field.
+
+Model Signature Types
+---------------------
+
+MLflow supports two primary types of signatures: a column-based signature for tabular-based data, and a tensor-based signature for tensor data. 
+
+Column-based Signatures
+^^^^^^^^^^^^^^^^^^^^^^^
+Column-based signatures are commonly utilized in traditional machine learning models that require tabular data inputs, such as a Pandas DataFrame. 
+These signatures consist of a series of columns, each potentially having a name and a specified data type. The type of each column in both input 
+and output corresponds to one of the :ref:`supported data types <supported-data-types-column>`, and columns can be optionally named. Additionally, 
+columns in the input schema can be designated as ``optional``, indicating that their inclusion in the model's input is not mandatory and they can be 
+left out if necessary (see :ref:`Optional Column <optional-column>` for more details).
+
+
+.. _supported-data-types-column:
+
+Supported Data Types
+~~~~~~~~~~~~~~~~~~~~
+
+Column-based signatures support data primitives defined within the :py:class:`MLflow DataType <mlflow.types.DataType>` specification:
+
+* string
+* integer<sup>1</sup>
```

Comment:
It seems <sup> tag is not working

Reply:
dang forgot that reST has a special handler for superscript. Updated it.